pymongo/collection.py

```diff
@@ -554,15 +554,9 @@ def _legacy_write(self, sock_info, name, cmd, op_id,
 
     def _insert_one(
             self, doc, ordered,
-            check_keys, manipulate, write_concern, op_id, bypass_doc_val,
+            check_keys, write_concern, op_id, bypass_doc_val,
             session):
         """Internal helper for inserting a single document."""
-        if manipulate:
-            doc = self.__database._apply_incoming_manipulators(doc, self)
-            if not isinstance(doc, RawBSONDocument) and '_id' not in doc:
-                doc['_id'] = ObjectId()
```

Comment:
I guess this was just here for legacy insert()?

Reply:
Yep, the non-legacy `insert_one` method always adds the `_id` field before calling this method.